plugins/serializers/prometheusremotewrite/prometheusremotewrite.go

```diff
@@ -239,8 +236,10 @@ func (s *Serializer) SerializeBatch(metrics []telegraf.Metric) ([]byte, error) {
 		return nil, fmt.Errorf("unable to marshal protobuf: %w", err)
 	}
 	encoded := snappy.Encode(nil, data)
-	buf.Write(encoded) //nolint:revive // from buffer.go: "err is always nil"
-	return buf.Bytes(), nil
+	s.buf.Write(encoded) //nolint:revive // from buffer.go: "err is always nil"
+
+	out := make([]byte, 0, s.buf.Len())
+	return append(out, s.buf.Bytes()...), nil
```

Comment:
Hmmm you are doing the alloc here with the length of the plugin-global buffer but otherwise scratch on that buffer. How about only storing the _maximal-allocated-buffer-size_ in the plugin struct and alloc that locally. Wouldn't that yield the same improvement as line 241 will also alloc once?

Even better, you will not introduce a state (that requires locking) and you don't need to copy as you can directly write to the allocated `out` buffer...